### PR TITLE
add db init script for oracle and custom sequence name for undo log

### LINKED ISSUE #1639: custom undo_log sequence name in oracle
## Why you need it?
 Is your feature request related to a problem? Please describe in details
For some reasons, two micro servies share one Oracle database currently, i want to specify sequence diff undo_log table and sequence.

## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.
i want to custom sequence name in file.conf, something like 
transaction {
  undo.log.sequence = "undo_log_seq"
}
and the default value is undo-log table name concated "_seq"
compible with the current implenetation

## Other related information
according to ConfigurationKeys and UndoLogManagerOracle, currently is hard coded sequence name of "undo_log_seq";

### PR DESCRIPTION
**Pull request by tq02ksu**
### Ⅰ. Describe what this PR did
1. add db init script for oracle.
2. add oracle jdbc driver to seata server
3. feature:  custom sequence name for undo log

### Ⅱ. Does this pull request fix one issue?

fixes #1639 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
the server can startup corrently with the script file

### Ⅳ. Describe how to verify it
exec sql file in db, startup the server

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -374,6 +374,11 @@
                 <artifactId>mysql-connector-java</artifactId>
                 <version>${mysql.client.version}</version>
             </dependency>
+            <dependency>
```
Comment (slievrly) [inline bom/pom.xml]: should also add in all.pom

File: pom.xml [hunk 2]
```diff
@@ -463,4 +463,11 @@
         </plugins>
     </build>
 
+    <repositories>
+        <repository>
+            <id>ICM</id>
+            <url>http://maven.icm.edu.pl/artifactory/repo/</url>
+            <releases><enabled>true</enabled></releases>
```
Comment (slievrly) [inline pom.xml]: format

File: server/src/main/resources/db_store_oracle.sql [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    on global_table (gmt_modified, status)
+/
+create index idx_transaction_id
+    on global_table (transaction_id)
+/
+
+-- the table to store BranchSession data
+create table branch_table (
+        branch_id number(20) not null,
+        xid varchar2(128) not null,
```
Comment (slievrly) [inline server/src/main/resources/db_store_oracle.sql]: format

File: server/src/main/resources/file.conf [hunk 4]
```diff
@@ -118,6 +118,8 @@ transaction {
   #schedule delete expired undo_log in milliseconds
   undo.log.delete.period = 86400000
   undo.log.table = "undo_log"
+  # sequence name, optional, default is table name + "_SEQ", case insensitive
+  undo.log.sequence = "undo_log_seq"
```
Comment (slievrly) [inline server/src/main/resources/file.conf]: undo_log table always create in client side.

Comment (tq02ksu) [reply server/src/main/resources/file.conf]: whatever the undo_log and its related sequence configuration and description write one place?

File: all/pom.xml [hunk 5]
```diff
@@ -420,6 +420,11 @@
             <artifactId>mysql-connector-java</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
```
Comment (slievrly) [inline all/pom.xml]: remove dependency. There is no direct dependency on ojdbc in the code. If the business uses oracle, it must be dependent on ojdbc in his pom.

**Comment by slievrly** [comment]
@tq02ksu please add oracle script to https://github.com/seata/seata/tree/develop/script

**Comment by a364176773** [comment]
Resolve code conflicts

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中
由于apache规范要求，我们不能引入oracle的driver
Due to the apache specification, we can't bring in oracle's driver.

**Pull request closed without merging**